include/dylib.hpp

```diff
@@ -276,20 +232,47 @@ class dylib {
         return dlerror();
     }
 #endif
+    void open(const std::string &path, const std::string &name, bool decorations) {
+        std::string final_name = name;
+        std::string final_path = path;
+
+        if (decorations)
+            final_name = prefix + name + extension;
+
+        if (path != dylib::os_library_paths && path.find_last_of('/') != path.size() - 1)
+            final_path += '/';
+
+        m_handle = _open((final_path + final_name).c_str());
+
+        if (!m_handle)
+            throw handle_error(get_handle_error(final_path + final_name));
+    }
+
+    void close() noexcept {
+        if (m_handle) {
+            _close(m_handle);
+            m_handle = nullptr;
```

Comment:
No need.

Reply:
And if you adopt my earlier comments, the handle is never null.

Reply:
Done